pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -50,9 +41,60 @@ function format-data-device() {
 format-data-device
 {{- end }}
 
-{{ range $name, $image := .images -}}
-docker-preload "{{ $name }}" "{{ $image }}"
-{{ end }}
+LAST_DOWNLOADED_HYPERKUBE_IMAGE=""
+if [[ -f "$PATH_LAST_DOWNLOADED_HYPERKUBE_IMAGE" ]]; then
+  LAST_DOWNLOADED_HYPERKUBE_IMAGE="$(cat "$PATH_LAST_DOWNLOADED_HYPERKUBE_IMAGE")"
+fi
+
+LAST_COPIED_HYPERKUBE_IMAGE=""
+if [[ -f "$PATH_LAST_COPIED_HYPERKUBE_IMAGE" ]]; then
+  LAST_COPIED_HYPERKUBE_IMAGE="$(cat "$PATH_LAST_COPIED_HYPERKUBE_IMAGE")"
+fi
+
+echo "Checking whether we need to preload a new hyperkube image..."
+if [[ "$LAST_DOWNLOADED_HYPERKUBE_IMAGE" != "{{ .hyperkubeImage }}" ]]; then
+  echo "Preloading hyperkube image ({{ .hyperkubeImage }}) because last downloaded image ($LAST_DOWNLOADED_HYPERKUBE_IMAGE) is outdated"
+  {{ .pathDockerBinary }} pull "{{ .hyperkubeImage }}"
+
+  echo "Starting temporary hyperkube container to copy binaries to host"
+
+{{- if semverCompare "< 1.17" .kubernetesVersion }}
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw "{{ .hyperkubeImage }}" /bin/sh -c "cp /usr/local/bin/kubelet $PATH_HYPERKUBE_DOWNLOADS"
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw "{{ .hyperkubeImage }}" /bin/sh -c "cp /usr/local/bin/kubectl $PATH_HYPERKUBE_DOWNLOADS"
+{{- else if semverCompare "< 1.19" .kubernetesVersion }}
```

Comment:
I know this is legacy code, but Isn't it possible the command used for 1.17 and 1.18 to be used also for <1.17?
```suggestion
{{- if semverCompare "< 1.19" .kubernetesVersion }}
```

Locally I have no issues with both
```
docker run --rm -v /tmp/kubelet:/tmp/kubelet:rw --entrypoint /bin/sh k8s.gcr.io/hyperkube:v1.16.10 -c "cp /usr/local/bin/kubelet /tmp/kubelet"
docker run --rm -v /tmp/kubelet:/tmp/kubelet:rw --entrypoint /bin/sh k8s.gcr.io/hyperkube:v1.17.10 -c "cp /usr/local/bin/kubelet /tmp/kubelet"
```

to take the kubelet binary.

Reply:
Maybe, but let's not touch it now. It was like this before and it works well enough.

Reply:
Ok.